src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -1841,5 +1855,202 @@ string[] linkingTargetFields
+            DisplayName = "Valid rest operations specified in rest methods for stored procedure pass config validation.")]
+        public void ValidateRestMethodsForEntityInConfig(
+            EntitySourceType sourceType,
+            string methods,
+            bool exceptionExpected,
+            string expectedErrorMessage = "")
+        {
+            string runtimeConfigString = @"{
+                    " +
+                @"""$schema"": ""test_schema""," +
+                @"""data-source"": {
+                    ""database-type"": ""mssql"",
+                    ""connection-string"": ""testconnectionstring"",
+                    ""options"":{
+                        ""set-session-context"": false
+                    }
+                },
+                ""runtime"": {
+                    ""host"": {
+                    ""mode"": ""development"",
+                    ""authentication"": {
+                        ""provider"": ""StaticWebApps""
+                    }
+                  },
+                  ""rest"": {
+                    ""enabled"": true,
+                    ""path"": ""/api""
+                    },
+                  ""graphql"": {
+                       ""enabled"": true,
+                       ""path"": ""/graphql"",
+                       ""allow-introspection"": true
+                    }
+                },
+                ""entities"": {
+                    ""HybridEntity"":{
+                        ""source"": {
+                            ""object"": ""hybridSource"",
+                            ""type"":" + $"\"{sourceType}\"" + @"
+                        },
+                        ""permissions"": [
+                           {
+                            ""role"": ""anonymous"",
+                            ""actions"": [
+                               ""*""
+                            ]
+                           }
+                         ],
+                        ""rest"":{
+                            ""methods"":" + $"{methods}" + @"
+                         }
+                       }
+                    }
+                }";
+
+            RuntimeConfigLoader.TryParseConfig(runtimeConfigString, out RuntimeConfig runtimeConfig);
+
+            // Perform validation on the entity in the config and assert the expected results.
+            if (exceptionExpected)
+            {
+                DataApiBuilderException ex =
+                    Assert.ThrowsException<DataApiBuilderException>(() => RuntimeConfigValidator.ValidateEntityConfiguration(runtimeConfig));
+                Assert.AreEqual(expectedErrorMessage, ex.Message);
+                Assert.AreEqual(HttpStatusCode.ServiceUnavailable, ex.StatusCode);
+                Assert.AreEqual(DataApiBuilderException.SubStatusCodes.ConfigValidationError, ex.SubStatusCode);
+            }
+            else
+            {
+                RuntimeConfigValidator.ValidateEntityConfiguration(runtimeConfig);
```

Comment:
Should there be an assert here? This being false would be unexpected and may still fail the test. However, would the logging of why the test failed give enough information to troubleshoot? e.g. `Assert(expected, actual, failureMessage).`

Reply:
`ValidateEntityConfiguration` method is of void return type just like other methods in runtimeconfigvalidator `ValidatePermissionsInConfig`.